snuba/subscriptions/data.py

```diff
@@ -223,3 +233,40 @@ def to_dict(self) -> Mapping[str, Any]:
 class Subscription(NamedTuple):
     identifier: SubscriptionIdentifier
     data: SubscriptionData
+
+
+class SubscriptionWithTick(NamedTuple):
+    subscription: Subscription
+    tick: Tick
+
+
+@dataclass(frozen=True)
+class ScheduledSubscriptionTask:
+    """
+    A scheduled task represents a unit of work (a task) that is intended to
+    be executed at (or around) a specific time.
+    """
+
+    # The time that this task was scheduled to execute.
+    timestamp: datetime
+
+    # The task that should be executed.
+    task: SubscriptionWithTick
+
+
+class SubscriptionScheduler(ABC):
+    """
+    The scheduler maintains the scheduling state for subscription tasks and
+    provides the ability to query the schedule to find tasks that were
+    scheduled between the time interval of a tick.
```

Comment:
why `to find tasks that were scheduled between the time interval of a tick` ?
Should this say `find tasks that should be scheduled between ....` 
If I am getting this right you are using this to schedule subscription, not to fetch what was already scheduled

Reply:
Yea it was sloppy wording, will update